src/ipa_pc/mod.rs

```diff
@@ -678,14 +678,14 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         })
     }
 
-    fn check<'a, R: RngCore>(
+    fn check<'a>(
         vk: &Self::VerifierKey,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
         point: G::ScalarField,
         values: impl IntoIterator<Item = G::ScalarField>,
         proof: &Self::Proof,
         opening_challenge: G::ScalarField,
-        _rng: &mut R,
+        _rng: Option<&mut dyn RngCore>,
```

Comment:
```suggestion
        _rng: impl Into<Option<&mut dyn RngCore>>,
```

Reply:
Why pass in an `_rng` that never gets used?

Reply:
This particular PC scheme doesn’t use an rng, but another one might, and so the trait is written to be flexible enough to accommodate both of them.

Reply:
Also, is there a need for the trait to be object safe or is `dyn` only being used avoid specifying a type when passing `None`?

Reply:
the latter

Reply:
Another approach is simply to pass an `RngCore` but provide a `PanicRng` that simply panics, but not sure that's any better since `dyn RngCore` looks efficient enough.

Reply:
Yeah I have somewhere an `OptionalRng` that panics if it's set to None, maybe that's useful here. Either way `dyn RngCore` is almost certainly not the bottleneck in these applications